src/Microsoft.AspNetCore.Razor.Evolution/RazorEngineBuilderExtensions.cs

```diff
@@ -26,6 +27,24 @@ public static IRazorEngineBuilder AddDirective(this IRazorEngineBuilder builder,
             return builder;
         }
 
+        public static IRazorEngineBuilder AddTargetExtension(this IRazorEngineBuilder builder, IRuntimeTargetExtension extension)
```

Comment:
I'm starting to think that "Target" might not be the best name. I fail at picking names but `RuntimeTarget`/`AddTargetExtension` don't make me think of "here's a place to tie into code generation"